packages/shared/src/components/auth/AuthOptions.tsx

```diff
@@ -239,70 +238,67 @@ function AuthOptions({
     onSetActiveDisplay(AuthDisplay.CodeVerification);
   };
 
-  useWindowEvents<SocialRegistrationFlow>(
-    'message',
-    AuthEvent.SocialRegistration,
-    async (e) => {
-      if (e.data?.flow) {
-        const connected = await getKratosFlow(
-          AuthFlow.Registration,
-          e.data.flow,
-        );
+  useEventListener(globalThis, 'message', async (e) => {
+    if (e.data?.eventKey !== AuthEvent.SocialRegistration) {
+      return undefined;
+    }
 
-        trackEvent({
-          event_name: AuthEventNames.RegistrationError,
-          extra: JSON.stringify({
-            error: {
-              flowId: connected?.id,
-              messages: connected?.ui?.messages,
-            },
-            origin: 'window registration flow error',
-          }),
-        });
+    if (e.data?.flow) {
+      const connected = await getKratosFlow(AuthFlow.Registration, e.data.flow);
 
-        if (
-          [4010002, 4010003, 4000007].includes(connected?.ui?.messages?.[0]?.id)
-        ) {
-          const registerUser = {
-            name: getNodeValue('traits.name', connected.ui.nodes),
-            email: getNodeValue('traits.email', connected.ui.nodes),
-            image: getNodeValue('traits.image', connected.ui.nodes),
-          };
-          const { result } = await getKratosProviders(connected.id);
-          setIsConnected(true);
-          await onSignBackLogin(registerUser, result[0] as SignBackProvider);
-          return onSetActiveDisplay(AuthDisplay.SignBack);
-        }
+      trackEvent({
+        event_name: AuthEventNames.RegistrationError,
+        extra: JSON.stringify({
+          error: {
+            flowId: connected?.id,
+            messages: connected?.ui?.messages,
+          },
+          origin: 'window registration flow error',
+        }),
+      });
 
-        return displayToast(labels.auth.error.generic);
-      }
-      const bootResponse = await refetchBoot();
-      if (!bootResponse.data.user || !('email' in bootResponse.data.user)) {
-        trackEvent({
-          event_name: AuthEventNames.SubmitSignUpFormError,
-          extra: JSON.stringify({
-            error: 'Could not find email on social registration',
-          }),
-        });
-        return displayToast(labels.auth.error.generic);
+      if (
+        [4010002, 4010003, 4000007].includes(connected?.ui?.messages?.[0]?.id)
```

Comment:
What do these magic numbers mean? Should we add const variable with more meaningful names?

Reply:
These are error codes from Kratos. We should have an enum to hold these values `KRATOS_ERROR`.

Reply:
Yeah I could add the enum here 👌 @sshanzel do you know from where are those codes taken/documented?

Reply:
Should be available through this document: sendAudioData

Reply:
Added in 52ab383713c09115c39c2380ed2304d8ba290c11